OpenUpSA/wazimap-ng-ui
# data whitelisting

## Description
`restrict_values` works across the app now : 
* rich data panel
   - charts show only the whitelisted sub-indicators
   - filter dropdown options are whitelisted
* data mapper
   - data mapper shows only the whitelisted sub-indicators
   - choropleth shows only the whitelisted sub-indicators
   - filter dropdown options are whitelisted

we skipped the my view : https://openupsa.slack.com/archives/C024AJAJ3T8/p1687203284338419

## Related Issue
https://wazimap.atlassian.net/browse/WNCM-744

## How is it tested automatically?
added test steps into `__tests__/gui/data_whitelist.feature`

## How should a reviewer test it locally
* add `restrict_values` to profile config
* confirm that in the web page data mapper, rich data charts and filter dropdowns are filtered out to show just the restricted values
* confirm that in the data mapper and the rich data panel, the sub-indicators that are not included in `restrict_values` are not displayed in anywhere else
* confirm that in the data mapper and the rich data panel if none of the sub-indicators of an indicator exist in `restrict_values`, the indicator itself is not visible

## Screenshots
![image](https://github.com/OpenUpSA/wazimap-ng-ui/assets/53019884/6d5a2bdd-c2e9-4fca-9b9c-660ad84cd081)


## Changelog

### Added

### Updated
* `src/js/api.js` to add wrapper functions. these functions call the functions that fetch the API data, then filter them out depending on the config

### Removed


## Checklist


### Pull Request


### Code Quality


### Testing
